### ISSUE TITLE
Create Post button available even without permission to post

### ISSUE THREAD

**Issue by MoshiKoi** [codidact/qpixel#994]
Even when users don't have permission to post in a category, the create post button is still shown

![Blog category with "Create Post" button](https://user-images.githubusercontent.com/54333972/215688030-90137cb8-8e36-49a8-baf5-d1b2946cbd88.png)

Clicking on it gives the following notice:

![Notice: You don't have a high enough trust level to post in the Blog category.](https://user-images.githubusercontent.com/54333972/215688286-a8299b68-e380-4c91-a357-8a4fbbf05eeb.png)

This just leaves a bad user experience with a button leading to a no-access page, so it should probably not be shown to unpermitted users.

**Comment by trichoplax** [comment]
Options as I haven't yet formed opinions:
- Display the button.
    - When clicked redirect to a dead-end page (current behaviour).
    - When clicked redirect to a page explaining who can post to this category.
    - When clicked display a message on the same page.
- Display the button but disabled.
    - Have title text (tooltip/hovertext) that explains why (useless on mobile).
    - Have a message appear on click that explains why (perhaps in addition to having title text).
    - Have the name of the button be different when disabled to explain why.
- Don't display the button.
    - Display nothing.
    - Display a link to a help guide on who can post where.

**Comment by trichoplax** [comment]
I do have one opinion:

> You don't have a high enough trust level

This sounds more personally directed and accusatory than it needs to. Wherever it is decided to display an explanation, I'd prefer it to be something more neutral. Something like:

> The Blogger ability is required to post in this category

(For whatever real ability it happens to be.)

It would also be useful for the ability name to be a link to more information about how it is earned or who gets it (if it's something most people won't earn, like moderator or admin).

**Comment by cellio** [comment]
> Have the name of the button be different when disabled to explain why.

I like this approach because it's accessible on all platforms.  Can we think of wording concise enough to fit there?  Does it work to use something like "posting restricted", and if you click on it you get one of those in-page messages (like we do for "thanks for your flag") that explains more?

**Comment by trichoplax** [comment]
Would phrasing it as a question make it more clear that it can be clicked on for more info? Instead of:

> Posting restricted

We could have:

> Why can't I post?

It's still longer than the existing:

> Create post

but it might just fit in the horizontal space available on a phone screen. I can't test at present as I'm not at a computer.

My only slight concern is that it puts a slightly negative message at the top of every blog post. I'm currently leaning towards this rather than lack of info though.

**Comment by cellio** [comment]
The more I think about it the more uncertain I become (and I was already uncertain).  Let's ask our community for feedback: https://meta.codidact.com/posts/287836

**Comment by trichoplax** [comment]
Good idea. Now that I'm back at a computer, here are some button wordings on a typical phone screen. Those that are too long give a horizontal scroll bar in the header area, which I'm assuming we want to avoid.

![Existing button wording "Create Post"](https://meta.codidact.com/uploads/ul42xpwtcc5klx036puqlhpaj9h2)

![Alternative button wording "Posting restricted"](https://meta.codidact.com/uploads/xknmti1ujp2i9wzwdisecwf1o2s5)

![Alternative button wording "Why can't I post?"](https://meta.codidact.com/uploads/h8mhcq2wyihal94xwpleu4b00nr6)

![Alternative button wording "Can't post?"](https://meta.codidact.com/uploads/lfi4zzevnje8l8jf6gdbj4fi66eb)

Bear in mind that there may be slightly more room for wording if we make category subsections tabs rather than buttons as suggested in [Confusing reuse of styling between button and tab header](https://meta.codidact.com/posts/287669).
